fivetran_provider_async/hooks.py

```diff
@@ -372,56 +381,107 @@ def get_last_sync(self, connector_id: str, xcom: str = "") -> pendulum.DateTime:
 
         :param connector_id: Fivetran connector_id, found in connector settings
             page in the Fivetran user interface.
-        :param previous_completed_at: The last time the connector ran, collected on Sensor
-            initialization.
-        :param reschedule_wait_time: Optional, if connector is in reset state
-            number of seconds to wait before restarting, else Fivetran suggestion used
-        :param reschedule_time: Deprecated
+        :param completed_after_time: The time we are comparing the Fivetran
+            `succeeded_at` and `failed_at` timestamps to. The method returns
+            True when both the last `succeeded_at` exceeds the
+            `completed_after_time` and other conditions (determined by other
+            params to this method) are met.
+        :param reschedule_wait_time: Optional. If connector is in a
+            rescheduled state, this will be the number of seconds to wait
+            before restarting. If None, then Fivetran's suggestion is used
+            instead.
+        :param always_wait_when_syncing: If True, then this method will
+            always return False when the connector's sync_state is "syncing",
+            no matter what.
+        :param propagate_failures_forward: If True, then this method will always
+            raise an AirflowException when the most recent connector status is
+            a failure and there are no successes dated after the target time.
+            Specifically, this makes it so that
+            `completed_after_time > failed_at > succeeded_at` is considered a
+            fail condition.
         """
-        if reschedule_time is not None:
-            import warnings
-
-            warnings.warn(
-                "`reschedule_time` arg is deprecated. Please use `reschedule_wait_time` instead.",
-                stacklevel=2,
-            )
-            if reschedule_wait_time is None:
-                reschedule_wait_time = reschedule_time
-
         # @todo Need logic here to tell if the sync is not running at all and not
         # likely to run in the near future.
         connector_details = self.get_connector(connector_id)
+        return self._determine_if_synced_from_connector_details(
+            connector_id=connector_id,
+            connector_details=connector_details,
+            completed_after_time=completed_after_time,
+            reschedule_wait_time=reschedule_wait_time,
+            always_wait_when_syncing=always_wait_when_syncing,
+            propagate_failures_forward=propagate_failures_forward,
+        )
+
+    def _determine_if_synced_from_connector_details(
+        self,
+        connector_id: str,
+        connector_details: dict[str, Any],
+        completed_after_time: datetime,
+        reschedule_wait_time: int | None = None,
+        always_wait_when_syncing: bool = False,
+        propagate_failures_forward: bool = True,
+    ):
```

Comment:
missing type annotation
```suggestion
    ) -> bool:
```